swap/swap.go

```diff
@@ -434,86 +436,86 @@ func (s *Swap) Balances() (map[enode.ID]int64, error) {
 	return balances, nil
 }
 
-// SentCheque returns the last sent cheque for a given peer
-func (s *Swap) SentCheque(peer enode.ID) (cheque *Cheque, err error) {
-	if swapPeer := s.getPeer(peer); swapPeer != nil {
-		return swapPeer.getLastSentCheque(), nil
-	}
-	err = s.store.Get(sentChequeKey(peer), &cheque)
-	return cheque, err
-}
-
-// SentCheques returns the last sent cheques for all known SWAP peers
-func (s *Swap) SentCheques() (map[enode.ID]*Cheque, error) {
-	cheques := make(map[enode.ID]*Cheque)
+// Cheques returns all known last sent and received cheques, grouped by peer
+func (s *Swap) Cheques() (map[enode.ID]map[string]*Cheque, error) {
+	cheques := make(map[enode.ID]map[string]*Cheque)
 
+	// get peer cheques from memory
 	s.peersLock.Lock()
 	for peer, swapPeer := range s.peers {
 		swapPeer.lock.Lock()
-		cheques[peer] = swapPeer.getLastSentCheque()
+		cheques[peer] = make(map[string]*Cheque)
```

Comment:
can you explain how that would happen?

Reply:
i don't think this is possible, since the `cheques` map has just been initialized on line `468`, and each `peer` from `s.peers` is appearing for the first time at this point.

the check in `addStoreCheques` is precisely not to override this newly generated maps from SWAP peers present in memory (which should always be more updated than their store images, should they exist).